third_party/move/move-vm/runtime/src/storage/struct_name_index_map.rs

```diff
@@ -42,59 +42,58 @@ impl StructNameIndexMap {
     /// if A == B, they have the same indices.
     pub(crate) fn struct_name_to_idx(
         &self,
-        struct_name: StructIdentifier,
+        struct_name: &StructIdentifier,
     ) -> PartialVMResult<StructNameIndex> {
-        // Note that we take a write lock here once, instead of (*): taking a read lock, checking
-        // if the index is cached, re-acquiring the (write) lock, and checking again, as it makes
-        // things faster.
-        // Note that even if we do (*), we MUST check if another thread has cached the index before
-        // we reached this point for correctness. If we do not do this, we can end up evicting the
-        // previous index, and end up with multiple indices corresponding to the same struct. As
-        // indices are stored inside types, type comparison breaks!
-        let mut index_map = self.0.write();
-
-        // Index is cached, return early.
-        if let Some(idx) = index_map.forward_map.get(&struct_name) {
-            return Ok(StructNameIndex(*idx));
+        {
+            let index_map = self.0.read();
```

Comment:
I think more idiomatic way is:
```rust
 if let Some(idx) =  self.0.read().forward_map.get(struct_name) {
   // Locked only here.
    return Ok(StructNameIndex(*idx));
}
```
Then also no need for scoping (though its fine to have it to explicitly specify the locked area).

Reply:
I like explicit scoping for code clarity